File: apps/modernization-ui/src/pages/patient/profile/labReport/PatientLabReport.ts
type AssociatedWith = {
    id: string;
    condition: string;
    local: string;
};

type TestResult = {
    test: string;
    result: string | null;
};

type PatientLabReport = {
    report: string;
    receivedOn: Date;
    reportingFacility: string | null;
    orderingProvider: string | null;
    orderingFacility: string | null;
    collectedOn: Date | null;
    results: TestResult[];
    associatedWith: AssociatedWith[];
    programArea: string;
    jurisdiction: string;
    event: string;
};

export type { PatientLabReport, AssociatedWith, TestResult };

File: apps/modernization-api/src/main/java/gov/cdc/nbs/patient/labreport/PatientLabReport.java
package gov.cdc.nbs.patient.labreport;

import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;
import gov.cdc.nbs.demographics.name.DisplayableSimpleName;
import gov.cdc.nbs.patient.events.tests.ResultedTest;

public record PatientLabReport(
    @JsonProperty(required = true) String eventId,
    LocalDateTime receivedDate,
    String processingDecision,
    LocalDateTime collectedDate,
    List<ResultedTest> testResults,
    AssociatedInvestigation associatedInvestigation,
    String programArea,
    @JsonProperty(required = true) String jurisdiction,
    long id,
    String specimenSource,
    String reportingFacility,
    DisplayableSimpleName orderingProvider,
    String orderingFacility) {

  public record AssociatedInvestigation(
      String id,
      String condition,
      String local,
      String status) {
  }
}
